aui.views/src/AUI/View/AListView.cpp

```diff
@@ -133,59 +117,62 @@ void AListView::setModel(const _<IListModel<AString>>& model) {
 
     mObserver->setModel(model);
     if (model) {
-        connect(model->dataRemoved, [&] {
-            mSelectionModel.clear();
-        });
+        connect(model->dataRemoved, [&] { mSelectionModel.clear(); });
     }
 }
 
 void AListView::handleMousePressed(AListItem* item) {
+    if (!(AInput::isKeyDown(AInput::LCONTROL) || AInput::isKeyDown(AInput::RCONTROL)) || !mAllowMultipleSelection) {
+        clearSelectionInternal();
+    }
 
-    if (!AInput::isKeyDown(AInput::LCONTROL) || !mAllowMultipleSelection) {
-        clearSelection();
+    auto index = AListModelIndex(mContent->getIndex());
+    if (mSelectionModel.contains(index))
+    {
+        mSelectionModel.erase(index);
+        item->setSelected(false);
+    }
+    else
+    {
+        mSelectionModel << index;
+        item->setSelected(true);
     }
-    mSelectionModel << AListModelIndex(mContent->getIndex());
-    item->setSelected(true);
 
     emit selectionChanged(getSelectionModel());
 }
 
-void AListView::clearSelection() {
+void AListView::clearSelectionInternal() {
     for (auto& s : mSelectionModel) {
         _cast<AListItem>(mContent->getViews()[s.getRow()])->setSelected(false);
     }
 
     mSelectionModel.clear();
 }
 
-void AListView::handleMouseDoubleClicked(AListItem* item) {
-    emit itemDoubleClicked(mContent->getIndex());
-}
+void AListView::handleMouseDoubleClicked(AListItem* item) { emit itemDoubleClicked(mContent->getIndex()); }
 
-void AListView::insertItem(size_t at, const AString& value) {
-    mContent->addView(at, _new<AListItem>(value));
-}
+void AListView::insertItem(size_t at, const AString& value) { mContent->addView(at, _new<AListItem>(value)); }
 
 void AListView::updateItem(size_t at, const AString& value) {
     _cast<AListItem>(mContent->getViews()[at])->setText(value);
 }
 
-void AListView::removeItem(size_t at) {
-    mContent->removeView(at);
-}
+void AListView::removeItem(size_t at) { mContent->removeView(at); }
 
-void AListView::onDataCountChanged() {
-    AUI_NULLSAFE(AWindow::current())->flagUpdateLayout();
-}
+void AListView::onDataCountChanged() { AUI_NULLSAFE(AWindow::current())->flagUpdateLayout(); }
 
-void AListView::onDataChanged() {
-    redraw();
-}
+void AListView::onDataChanged() { redraw(); }
 
 void AListView::selectItem(size_t i) {
```

Comment:
This way selectItem works differently on single selection and multiple selection lists. I think the default action should be default for both types of lists:
1. reset selection
2. select the desired item

As for your intention, I would suggest to introduce extra method to control the selection more precisely:

```
/**
 * @brief Selection action for updateSelectionOnItem.
 */
enum class SelectAction {
    /**
     * @brief Clears old selection and selects the specified index. Used by selectItem.
     */
    CLEAR_SELECTION_AND_SET,

    /**
     * @brief Selects the specified index. In single selection mode, acts like CLEAR_SELECTION_AND_SET.
     */
    SET,

    /**
     * @brief Deselects the specified index.
     */
    UNSET,


    /**
     * @brief Selects or deselects the specified index depending on it's current state.
     */
    TOGGLE,
};
void updateSelectionOnItem(size_t i, AListView::SelectAction action);
```

And update selectItem accordingly to avoid code duplication (in header):

```
/**
 * @brief Acts on i item as if the user were left-clicked without keyboard modifiers on i.
 */
void selectItem(size_t i) { updateSelectionOnItem(i, SelectAction::CLEAR_SELECTION_AND_SET) }
```

This way the PR changes are non-breaking for `selectItem`.

You may also refer to Qt: https://doc.qt.io/qt-6/qitemselectionmodel.html#SelectionFlag-enum